### PR TITLE
feat: update service worker when new version is detected

### LINKED ISSUE #3667: Update service worker when new version is detected
### Describe the problem

With #3412, SvelteKit can detect when new versions of the app have been deployed. It would be useful if the service worker was updated when that happens, if one exists.

### Describe the proposed solution

Inside the code that checks for updates, something like this:

```js
const registration = await navigator.serviceWorker.getRegistration(base || '/');

if (registration) {
  registration.update();
}
```

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dumbmoron**
this patch adds a function that tries to update the service worker (if it exists), and calls it when sveltekit detects that a new version has been deployed.
i'm not sure if this is the right place to put this -- i considered putting it in the body of check(), but it feels wrong to me. if you think it's a good idea, let me know and i'll update it.

closes #3667

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: packages/kit/src/runtime/client/client.js [hunk 1]
```diff
@@ -146,6 +146,19 @@ function native_navigation(url) {
 	return new Promise(() => {});
 }
 
+/**
+ * Checks whether a service worker is registered, and if it is,
+ * tries to update it.
+ */
+async function update_service_worker() {
+	if (BROWSER && 'serviceWorker' in navigator) {
```
Comment (eltigerchino) [inline packages/kit/src/runtime/client/client.js:154]:
```suggestion
	if ('serviceWorker' in navigator) {
```

**Comment by eltigerchino** [comment]
I don't have much knowledge about service workers but it looks good to me. cc: @dummdidumm what do you think?

File: packages/kit/src/runtime/client/client.js [hunk 2]
```diff
@@ -146,6 +146,19 @@ function native_navigation(url) {
 	return new Promise(() => {});
 }
 
+/**
+ * Checks whether a service worker is registered, and if it is,
+ * tries to update it.
+ */
+async function update_service_worker() {
+	if ('serviceWorker' in navigator) {
+		const registration = await navigator.serviceWorker.getRegistration(base || '/');
```
Comment (eltigerchino) [inline packages/kit/src/runtime/client/client.js:155]: > If this parameter is not provided, the current client's URL will be used by default.
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/getRegistration

Do we consider leaving the parameters empty?

**Review by eltigerchino — APPROVED** [review]

**Review by dummdidumm — APPROVED** [review]
Thank you!

**Pull request merged**